CoseSignTool.Tests/MainTests.cs

```diff
@@ -214,4 +214,175 @@ public void ReturnsHelpRequestedWhenNoOptionsAfterVerb()
+    {
+        string headersFile = FileSystemUtils.GenerateHeadersFile(@"[{""label"":""created-at"",""value"":"""",""protected"":false}]");
+        string payloadFile = FileSystemUtils.GeneratePayloadFile();
+
+        // sign with string headers
+        string[] args1 = ["sign", @"/p", payloadFile, @"/pfx", PrivateKeyCertFileChainedWithPassword, @"/pw", CertPassword, @"/ep", @"/sh", headersFile];
+        CoseSignTool.Main(args1).Should().Be((int)ExitCode.MissingRequiredOption, "Missing value in string headers file.");
+    }
+
+    [TestMethod]
+    public void SignWithDeserializationErrorStringHeaders()
+    {
+        string headersFile = FileSystemUtils.GenerateHeadersFile(@"[{""label"":""created-at"",""protected"":false}]");
+        string payloadFile = FileSystemUtils.GeneratePayloadFile();
+
+        // sign with string headers
+        string[] args1 = ["sign", @"/p", payloadFile, @"/pfx", PrivateKeyCertFileChainedWithPassword, @"/pw", CertPassword, @"/ep", @"/sh", headersFile];
+        CoseSignTool.Main(args1).Should().Be((int)ExitCode.UnknownError, "String headers file could not be deserialized.");
+    }
+
+    [TestMethod]
+    public void SignWithCommandLineIntAndStringHeaders()
+    {
+        string payloadFile = FileSystemUtils.GeneratePayloadFile();
+
+        // sign with int protected headers
+        string[] args1 = ["sign", @"/p", payloadFile, @"/pfx", PrivateKeyCertFileChainedWithPassword, @"/pw", CertPassword, @"/sph", "sph1=sphv1,sph2=sphv2", @"/iph", "iph1=12345,iph2=123"];
+        CoseSignTool.Main(args1).Should().Be((int)ExitCode.Success, "Signing with protected int headers in command line failed.");
+
+        // sign with int and string unprotected headers
+        args1 = ["sign", @"/p", payloadFile, @"/pfx", PrivateKeyCertFileChainedWithPassword, @"/pw", CertPassword, @"/suh", "suh1=testsigning,suh2=value2", @"/iuh", "iuh1=12345,iuh2=123"];
+        CoseSignTool.Main(args1).Should().Be((int)ExitCode.Success, "Signing with int and string unprotected headers in command line failed.");
+
+        // sign with string unprotected headers
+        args1 = ["sign", @"/p", payloadFile, @"/pfx", PrivateKeyCertFileChainedWithPassword, @"/pw", CertPassword, @"/suh", "suh3=testsigning,suh4=value2"];
+        CoseSignTool.Main(args1).Should().Be((int)ExitCode.Success, "Signing with unprotected string headers in command line failed.");
+
+        // sign with string protected headers
+        args1 = ["sign", @"/p", payloadFile, @"/pfx", PrivateKeyCertFileChainedWithPassword, @"/pw", CertPassword, @"/sph", "sph3=testsigning"];
+        CoseSignTool.Main(args1).Should().Be((int)ExitCode.Success, "Signing with protected string headers in command line failed.");
+    }
+
+    [TestMethod]
+    public void SignWithCommandLineProtectedAndUnprotectedHeaders()
+    {
+        string payloadFile = FileSystemUtils.GeneratePayloadFile();
+
+        // sign with string protected headers

```

Comment:
Is this to implement later? Or just a stub of something that got rolled into another test?